src/main/java/org/jenkinsci/remoting/engine/JnlpAgentEndpointResolver.java

```diff
@@ -312,6 +319,45 @@ public int compare(String o1, String o2) {
         return null;
     }
 
+    private boolean isPortVisible(String hostname, int port, int timeout) {
+        boolean exitStatus = false;
+        Socket s = null;
+        String reason = null;
+
+        try {
+            s = new Socket();
+            s.setReuseAddress(true);
+            SocketAddress sa = new InetSocketAddress(hostname, port);
+            s.connect(sa, timeout);
+        } catch (IOException e) {
+            if (e.getMessage().equals("Connection refused")) {
+                reason = "port " + port + " on " + hostname + " is closed.";
+            }
+            if (e instanceof UnknownHostException) {
+                reason = "node " + hostname + " is unresolved.";
+            }
+            if (e instanceof SocketTimeoutException) {
+                reason = "timeout while attempting to reach node " + hostname + " on port " + port;
+            }
+        } finally {
```

Comment:
maybe it makes sense to suppress other exceptions (runtime&Co)

Reply:
Yes. It make sense.